test/testTrainingFunction.php

```diff
@@ -0,0 +1,162 @@
+<?php
+require '../trainingFunction.php';
+use PHPUnit\Framework\TestCase;
+class testTrainingFunction extends TestCase {
+
+    public function testGetAllDatesSuccess_1(){
+        $db_array = [
+                [
+                    'date'=>'2022-07-10',
+                    'exercise'=>'Max Hang',
+                    'difficulty'=>'hard',
+                    'weight_added_kg'=>50,
+                    'comments'=>'blah blah blah'
+                ],
+                [
+                    'date'=>'2022-04-06',
+                    'exercise'=>'Repeater',
+                    'difficulty'=>'easy',
+                    'weight_added_kg'=>10,
+                    'comments'=>'these are some words'
+                ]
+            ];
+        $result = getAllDates($db_array);
+        $expected = ['2022-07-10', '2022-04-06'];
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testGetAllDatesFailure_1(){
```

Comment:
it would be good here to add a comment that explains what the failure is and what the graceful handling is